<?xml version="1.0"?>
<launch>

  <!--copied from linobot/bringup.launch /opt/ros/indigo/share/linorobot/launch/bringup.launch -->
  <!--serial communication between arduino and pc via usb /-->
  <node name="arduino_serial_node" pkg="rosserial_python" type="serial_node.py" output="screen">
    <param name="port" value="/dev/linobase" />
    <param name="baud" value="57600" />
  </node>


 <node pkg="ros_arduino_imu" type="raw_imu_bridge_node" name="raw_imu_bridge" output="screen" respawn="false">
   <!--default world_frame: eun-->
   <param name="imu/use_mag_msg" value="false"/> 
     <!--when use_mag_msg=false /imu/mag messages are sent in Vector3Stampe, true=MagneticField(depricated) -->
  <!--rosparam>
   imu/gyroscope_bias: {x: -0.013545, y: 0.056887, z: 0.012693 }
   imu/accelerometer_bias: {x: -0.422578, y: 0.053516, z: -0.310391 }
    </rosparam-->
   <param name="imu/perform_calibration" value="true" />
 </node>


  <!--node name="jbot_base_node" pkg="jazzybot" type="jbot_base_node" output="screen"-->
  <node pkg="jazzybot" type="jbot_base2_node" name="jbot_base2_node" output="screen">
    <!--published tf-->
    <param name="odom_frame" value="odom" />  
    <param name="baselink_frame" value="base_link" />

    <!--publised topics-->
    <param name="vel_topic" value="raw_vel" />
    <param name="wheel_odom_topic" value="/odometry/wheelodom"/> <!--new. Requres static transfirm to set parent to odom-->

    <!--subscribed topicais - requred -->
    <param name="encoder_topic" value="encoders"/>  <!--from arduion-->
    <param name="imu_topic" value="imu/data" />
    <!-- /cmd_vel - id hardcoded -->

    <!--other misc settings-->
    <param name="odom_publish_rate" value="40"/>
    <param name="raw_pwm_pub_hz" value="6"/> <!--new. Rate in hz at which to update pwm value sent to wheel motor hardware-->
    <!--param name="debug_level" value="0"/--> <!--default is 0/off--> 
  </node>




  <!--http://wiki.ros.org/imu_filter_madgwick-->
  <!-- world_frame default: "nwu" ?? -->
  <!-- reaures mag_msg = true in ros_arduino_imu -->
  <!-- uncomment these-->
  <node pkg="imu_filter_madgwick" type="imu_filter_node" name="imu_filter_madgwick" output="screen" respawn="false" >
    <param name="world_frame" value="eun"/>  <!--eun is required by robot_localization/ekf_fusion-->
    <param name="use_magnetic_field_msg" value="false"/>
    <param name="publish_tf" value="true" /> <!--required true for robot_pose_ekf-->
    <param name="fixed_frame" value="imu0_link" /> <!--this is the published imu_link parent-->
    <!---param name="fixed_frame" value="imu_link" /--> <!--this is the published imu_link parent-->
    <param name="reverse_tf" value="false" />
    <param name="publish_debug_topics" value="true" />
  </node>


  <!--http://wiki.ros.org/imu_complementary_filter-->
  <!--param name="fixed_frame" value="imu_baselink" /-->
  <!--uses mag_msg and eun-->
  <!--node pkg="imu_complementary_filter" type="complementary_filter_node" name="imu_complementary_filter" output="screen" respawn="false" >
    <param name="fixed_frame" value="odom" />
    <param name="use_mag" value="true" />
    <param name="publish_tf" value="true"/>
    <param name="publish_debug_topics" value="true" />
    <param name="reverse_tf" value="false" />
  </node-->

  <!-- Publish a static coordinate transform frame (tf) using an x/y/z offset in meters and yaw/pitch/roll in radians. 
                               Last arg is The period, in milliseconds, specifies how often to send a transform. 100ms (10hz) is a good value. --> 
  <!--node pkg="tf" type="static_transform_publisher" name="base_link_to_imu_base_link" args="-.381 0 .343 0 0 0 base_link imu_baselink 100"/-->
  <node pkg="tf" type="static_transform_publisher" name="base_link_to_imu_base_link" args="-.381 0 .343 0 0 0 base_link imu0_link 100"/>
  <node pkg="tf" type="static_transform_publisher" name="odometry_to_baselink" args="0 0 0 0 0 0 odom  /odometry/wheelodom 100"/> 




</launch>

